packages/styles/src/components/dialog.scss

```diff
@@ -0,0 +1,246 @@
+  overscroll-behavior: contain;
+  border: 2px solid var(--post-contrast-color); // Ensures good contrast when bg is dark against dark backdrop
+
+  &::backdrop {
+    background-color: rgba(0, 0, 0, 0.8);
+    backdrop-filter: blur(0);
+  }
+
+  @include utilities.high-contrast-mode {
+    // Show the border in HCM
+    border-width: 2px;
+
+    // Mark the backdrop as inactive in HCM
+    &::backdrop {
+      background-image: linear-gradient(
+        135deg,
+        CanvasText 4.55%,
+        transparent 4.55%,
+        transparent 50%,
+        CanvasText 50%,
+        CanvasText 54.55%,
+        transparent 54.55%,
+        transparent 100%
+      );
+      background-size: 22px 22px;
+      backdrop-filter: none;
+      background-color: transparent;
+      forced-color-adjust: none;
+    }
+  }
+
+  // Sizes
+  // [small, medium (default), large]
+  &[data-size='small'] {
+    min-width: 296px;
+    max-width: 388px;
+  }
+
+  &[data-size='large'] {
+    min-width: min(600px, 90vw);
+    max-width: 792px;
+  }
+
+  // Positioning
+  // [top, center (default), bottom]
+  &[data-position='top'] {
+    top: 2rem;
+    bottom: auto;
+  }
+
+  &[data-position='bottom'] {
+    top: auto;
+    bottom: 2rem;
+  }
+}
+
+dialog > .dialog-grid {
+  margin: spacing.$size-regular spacing.$size-regular 0 spacing.$size-regular;
+  display: grid;
+  column-gap: spacing.$size-regular;
+  grid-template-columns: auto 1fr auto;
+  grid-template-areas:
+    'icon header close-button'
+    'icon body close-button'
+    'controls controls controls';
+
+  // Propagate bg color to the controls
+  background-color: inherit;
+}
+
+:where(
+    post-dialog-icon,
+    post-dialog-header,
+    post-dialog-body,
+    post-dialog-controls,
+    post-dialog-close
+  ):empty {
+  display: none;
+}
+
+post-dialog-icon {
+  grid-area: icon;
+  display: block;
+
+  > post-icon {
+    width: spacing.$size-big;
+    height: spacing.$size-big;
+
+    // Larger icon for bigger notification dialogs
+    dialog:not([size='small']) & {
+      @include media-breakpoint-up(rg) {
+        width: spacing.$size-small-huge;
+        height: spacing.$size-small-huge;
+      }
+    }
+  }
+}
+
+post-dialog-header {
+  grid-area: header;
+
+  > *:first-child {
+    margin-top: 0;
+  }
+}
+
+post-dialog-body {
+  grid-area: body;
+  margin-bottom: 0;
+
+  > *:last-child {
+    margin-bottom: 0;
+  }
+}
+
+post-dialog-controls {
+  grid-area: controls;
+  position: sticky;
+  bottom: 0;
+  padding-block: spacing.$size-regular;
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row-reverse;
+  gap: spacing.$size-regular;
+  background-color: inherit;
+
+  button {
+    @include media-breakpoint-down(sm) {
+      width: 100%;
+    }
+  }
+}
+
+post-dialog-close {
+  position: sticky;
+  top: 0;
+  grid-area: close-button;
+
+  > .btn-close {
+    width: spacing.$size-large;
+    height: spacing.$size-large;
+    min-height: 0;
+  }
+}
+
+// Animations
+// [slide-in, pop-in, none]
```

Comment:
Slide in feels more natural for dialogs that are positioned at the top or the bottom, they slide in from offscreen. For those positioned in the middle of the screen, slide in would cause too much motion, so a pop-in fits better. 

This might change once we think about motion holistically, but for the moment I'd prefer to keep it in like this.